File: codegen/nim/fixtures/resource/deliveries_api.nim
import jester, marshal



proc deliveriesGet*(req: PRequest) : tuple[code: HttpCode, content: string] =
  # Get a list of deliveries
  
  let respBody = ""
  result = (code: Http200, content: respBody)

proc deliveriesPost*(req: PRequest) : tuple[code: HttpCode, content: string] =
  # Create/request a new delivery
  
  let respBody = ""
  result = (code: Http200, content: respBody)

proc getDeliveriesByDeliveryID*(deliveryId: string, req: PRequest) : tuple[code: HttpCode, content: string] =
  # Get information on a specific delivery
  
  let respBody = ""
  result = (code: Http200, content: respBody)

proc deliveriesByDeliveryIdPatch*(deliveryId: string, req: PRequest) : tuple[code: HttpCode, content: string] =
  # Update the information on a specific delivery
  
  let respBody = ""
  result = (code: Http200, content: respBody)

proc deliveriesByDeliveryIdDelete*(deliveryId: string, req: PRequest) : tuple[code: HttpCode, content: string] =
  # Cancel a specific delivery
  
  let respBody = ""
  result = (code: Http200, content: respBody)

File: codegen/fixtures/server_resources/deliveries_api.py
import json as JSON
import jsonschema
from jsonschema import Draft4Validator
from flask import Blueprint, jsonify, request


import os
dir_path = os.path.dirname(os.path.realpath(__file__))


deliveries_api = Blueprint('deliveries_api', __name__)


@deliveries_api.route('/deliveries', methods=['GET'])
def deliveries_get():
    '''
    Get a list of deliveries
    It is handler for GET /deliveries
    '''
    
    return jsonify()


@deliveries_api.route('/deliveries', methods=['POST'])
def deliveries_post():
    '''
    Create/request a new delivery
    It is handler for POST /deliveries
    '''
    
    return jsonify()


@deliveries_api.route('/deliveries/<deliveryId>', methods=['GET'])
def getDeliveriesByDeliveryID(deliveryId):
    '''
    Get information on a specific delivery
    It is handler for GET /deliveries/<deliveryId>
    '''
    
    return jsonify()


@deliveries_api.route('/deliveries/<deliveryId>', methods=['PATCH'])
def deliveries_byDeliveryId_patch(deliveryId):
    '''
    Update the information on a specific delivery
    It is handler for PATCH /deliveries/<deliveryId>
    '''
    
    return jsonify()


@deliveries_api.route('/deliveries/<deliveryId>', methods=['DELETE'])
def deliveries_byDeliveryId_delete(deliveryId):
    '''
    Cancel a specific delivery
    It is handler for DELETE /deliveries/<deliveryId>
    '''
    
    return jsonify()
